pgfinder/pgio.py

```diff
@@ -68,17 +68,48 @@ def ftrs_reader(file: Union[str, Path]) -> pd.DataFrame:
         ff["Inferred structure"] = np.nan
         ff["Theo (Da)"] = np.nan
         # Renames columns to expected column heading required for data_analysis function
-        ff.rename(
-            columns={
-                "Id": "ID",
-                "chargeOrder": "Charge",
-                "apexRetentionTimeMinutes": "RT (min)",
-                "apexMwMonoisotopic": "Obs (Da)",
-                "maxIntensity": "Intensity",
-                "maxAveragineCorrelation": "corrMax",
-            },
-            inplace=True,
-        )
+        ftrs_52_columns = [
+            "Id",
+            "apexRetentionTime",
+            "charges",
+            "mwMonoIsotopicMass",
+            "apexIntensity",
+        ]
+
+        ftrs_311_columns = [
+            "Id",
+            "apexRetentionTimeMinutes",
+            "chargeOrder",
+            "apexMwMonoisotopic",
+            "maxIntensity",
+        ]
+
+        pgfinder_columns = [
+            "ID",
+            "RT (min)",
+            "Charge",
+            "Obs (Da)",
+            "Intensity",
+        ]
+
+        is_ftrs_52 = set(ftrs_52_columns).issubset(ff.columns)
```

Comment:
I think it might be able to reduce the code here and below by having...

```
is_ftrs = set(ftrs_52_coulumns).issubset(ff.columns) if is_ftrs_52 else set(ftrs_311_coulumns).issubset(ff.columns)
```

The subsequent `if: ... elif: ... else:` could then be a `try: except ValueError:` construct.

Reply:
You were right in thinking that this was a bit of future-proofing! It looked like the only way to test for 3 or more conditions (like 3.11, 5.2, *and* some unknown version — the `else` here) using `try` would have been to nest `try` blocks inside of the `except` blocks.

To avoid that nesting and keep everything flat (whilst writing some code with hopefully obvious variable names), I took the `if-elif-else` approach :)

Reply:
Now merged but whilst hiking on holiday my mind wandered back to this and there is a [design pattern](https://refactoring.guru/design-patterns/factory-method) for handling multiple options which is slightly more sophisticated than `if: ... elif: ... else: ...`.

Its a slightly cleaner way of structuring the code so it adheres to the ["O" principle of the SOLID](https://www.digitalocean.com/community/conceptual-articles/s-o-l-i-d-the-first-five-principles-of-object-oriented-design#open-closed-principle) (i.e. "*O*pen to extension but closed to modification").